component/main.jsonnet

```diff
@@ -69,16 +49,29 @@ local identityProviders = [
 
   idp {
     ldap+: {
-      ca: { name: authentication.RefName(idp.name) },
-      bindPassword: { name: authentication.RefName(idp.name) },
+      ca: { name: common.RefName(idp.name) },
+      bindPassword: {
+        name: if std.objectHas(idp, 'bindPasswordSecretRef') then params.secrets[idp.bindPasswordSecretRef] else common.RefName(idp.name),
+      },
     },
   }
   for idpname in std.objectFields(idps)
   if idps[idpname].type == 'LDAP'
+] + [
+  local idp = idps[idpname];
+  idp {
+    openID+: {
+      clientSecret: {
+        name: params.secrets[idp.openID.clientSecretRef],
+      },
+    },
+  }
+  for idpname in std.objectFields(idps)
+  if idps[idpname].type == 'OpenID'
```

Comment:
This is not necessary, and actually not wanted as the original idea behind Christian's suggestion was that we don't have to explicitly handle each IDP provider in the component (which the approach with `clientSecretRef` actually doesn't achieve), but provide a mechanism which automatically works with all IDP providers.

Just let the users configure `openID.clientSecret.name` to the name of a secret they're configuring with the secrets mechanism.

Reply:
Yeah, I got the same idea afterwards. It was friday evening ;)